### ISSUE TITLE
Stock/Commodity Lot Trade Syntax

### ISSUE THREAD

**Issue by MisterY** [simonmichael/hledger#1016]
In relation to #1015, I'd like to add an issue for the manual lot matching using common stock-trade scenario.
In ledger, I'm matching lots explicitly in order to get accurate capital gains calculation and reporting. Note that in the v3.1.3 version of ledger, the full syntax (including the lot purchase price *and* date) is required in order to make the full match on the lots. This, after the trade, removes them from the list that --lots option provide.

```
2019-01-05 Opening Balances
    Assets:Cash    1000.00 EUR
    Equity:Opening Balances    -1000.00 EUR

2019-01-10 Buy fund
    Assets:Shares    5 ETF @ 10 EUR
    Assets:Cash    -50 EUR

2019-01-15 Buy more
    Assets:Shares    5 ETF @ 12 EUR
    Assets:Cash    -60 EUR

2019-03-01 Sell
    Assets:Shares    -5 ETF {10 EUR} [2019-01-10] @ 15 EUR
    Assets:Shares    -1 ETF {12 EUR} [2019-01-15] @ 15 EUR
    Assets:Cash    90.00 EUR
    Capital Gains    -28.00 EUR
```

**Comment by simonmichael** [comment]
Thanks for the issues. Going to tweak the names to help me keep track of things..

**Comment by MisterY** [comment]
Absolutely. Wish I could be of more help in regards to writing code but it might be a while before I'm in a position to be productive in Haskell.

**Comment by simonmichael** [comment]
No time like the present! Making small tweaks to an existing project is the quickest way to learn Haskell. #hledger is ready to help.

**Comment by simonmichael** [comment]
Ledger lot syntax is preserved by `print` in.. master or 1.51, I forget which.
And it is operational and tested in the #2558 lots branch. So I think we could close this soon.

**Comment by simonmichael** [comment]
hledger 2.0pre1 handles the example above, with slight tweaks.
This declaration must be added to make `hledger check` pass (hledger tries to balance transactions both with and without cost basis/gains awareness; so it needs to know about gains accounts):
```
account Capital Gains  ; type:G
```

And this declaration must be added to make `hledger check lots` pass (or, cost basis annotations must be added to the buy transactions):
```
commodity 1.000 ETF    ; lots:
```

It's also possible to write the sell more simply. Full example:
```
account Capital Gains  ; type:G
commodity 1.000 ETF    ; lots:FIFO

2019-01-05 Opening Balances
    Assets:Cash    1000.00 EUR
    Equity:Opening Balances    -1000.00 EUR

2019-01-10 Buy fund
    Assets:Shares    5 ETF @ 10 EUR
    Assets:Cash    -50 EUR

2019-01-15 Buy more
    Assets:Shares    5 ETF @ 12 EUR
    Assets:Cash    -60 EUR

2019-03-01 Sell
    Assets:Shares   -6 ETF @ 15 EUR
    Assets:Cash     90.00 EUR
    Capital Gains  -28.00 EUR
```
```
$ hledger check lots
$ hledger print --lots --verbose-tags
...
```
